client/state/automated-transfer/package.json

```diff
@@ -0,0 +1,5 @@
+{
+	"sideEffects": [
+		"./init.js"
```

Comment:
this feels like a big thorn. `sideEffects: false` feels like an acceptable compromise since it enables an optimization, but tracking files with side effects feels somehow dirty and bug-prone. is there another way to resolve this? are we already manually labeling side-effecting files in other parts of Calypso?

Reply:
We do this also for CSS files exported by a package, for example.

The pure/impure flag is a property of the module and in an ideal world, it would be specified in the module source. Something like `"use strict"`. For a module, the flag says that it doesn't do anything observable during static initialization (at top level).

And just like `"use strict"`, the `"use (im)pure"` flag makes sense for an individual function, too, flagging it as side-effect-ful/free.

Reply:
+1 on what @jsnajdr said. Unfortunately, this is the only way we have of doing things right now.

In any case, the number of impure files to keep track of (at least in `state/`) should be rather small, so the tradeoff seems worth it.

Reply:
> We do this also for CSS files exported by a package, for example.

We specify a list of files that are impure in CSS? Or we use `sideEffects: False`?
It's more the "this governs everything" vs. "by the way you need to look here to know about parts of the system" that caught my attention - the locality of the change.

> Something like "use strict".

Is there a webpack plugin for this? That actually sounds like a great idea and one that wouldn't be too complicated.

Reply:
> > Something like "use strict".
> 
> Is there a webpack plugin for this? That actually sounds like a great idea and one that wouldn't be too complicated.

I've never come across such a plugin, and can't find one on npm.